storage/database.go

```diff
@@ -292,8 +308,13 @@ func (d *db) FetchBlocks(
 	starts []time.Time,
 ) ([]block.FetchBlockResult, error) {
 	callStart := d.nowFn()
-	n, err := d.namespaceFor(namespace)
+	if d.isOverloaded() {
```

Comment:
I think just in terms of separation of concerns and making the API more flexible and less coupled for this specific case it makes sense to do it at the layer above.

We can always add a reuseable RPC layer in between our current RPC implementation and the database in the future to alleviate any concerns about duplicating this logic down the line.